### ISSUE TITLE
Parse error: syntax error, unexpected end of file, expecting function (T_FUNCTION) or const (T_CONST) in mecha-cms.de/engine/kernel/state.php on line 1

### ISSUE THREAD

**Issue by AlexL777HRO** [mecha-cms/start#2]
Good morning, Tovic,

the installer trough /start.php going not done. My hosting is great prepared for the installation, but It's nothing possible over /start.php.

I have PHP 7.3 or 7.4 and an Apache web server at my hosting.

Can you help me with this?

Many thanks, Tovic :-)

**Comment by taufik-nurrohman** [comment]
The `.\engine\kernel\state.php` file should contain only the class definition. Also, why does it show an error on the first line? The first line is always a `<?php` opening tag.

Please paste the generated error log here in case it does exist.

**Comment by taufik-nurrohman** [comment]
Oh, could you try to disable the source code optimization and see if it works? It shows the error on the first line because the file content is compressed into a single line.

**Comment by AlexL777HRO** [comment]
Hi,

I'm trying to install in an SSD hosting too, but that errors are visible:

`✘ Could not pull mecha-cms/mecha due to network error.

✔ It looks like you are using a LiteSpeed web server. This web server is usually compatible with Apache web server configuration.

✔ Minimum PHP version required is 7.3.0. Your current PHP version is 7.4.33.

✔ PHP [dom](https://mecha.officiis.biz/start.php) extension is enabled.

✔ PHP [json](https://mecha.officiis.biz/start.php) extension is enabled.

✔ PHP [mbstring](https://mecha.officiis.biz/start.php) extension is enabled.`

Always a network error and the installer without code optimizing still not worked.

**Comment by taufik-nurrohman** [comment]
Make sure [`allow_url_fopen`](https://www.php.net/manual/en/filesystem.configuration.php#ini.allow-url-fopen) option is enabled.

**Comment by AlexL777HRO** [comment]
PHP is 7.3.x, allow url fopen is active and I have this error:

`✘ Could not pull mecha-cms/mecha due to network error.`

It's always only that error I show at Firefox browser under https://mecha.officiis.biz . I want to move, if the installation it's done, to the main domain under https://mecha-cms.de .

**Comment by taufik-nurrohman** [comment]
Must be from my side. Why not try to install it the manual way? There&rsquo;s an experimental website available to visit here: https://dev.mecha-cms.com. You can download the ZIP file directly from the home page.

The `dev.` prefix will be removed once everything is stable.

**Comment by AlexL777HRO** [comment]
Parse error: syntax error, unexpected end of file, expecting function (T_FUNCTION) or const (T_CONST) in mecha.officiis.biz/engine/kernel/state.php on line 1

So now, I'm trying with the code optimization. It's the same error, I think.

**Comment by taufik-nurrohman** [comment]
I need it without the code optimization.

**Comment by AlexL777HRO** [comment]
Ah, ok, I want always to install my CMS via browser way :-) It's ok, I want to try this now. Wait, please just a moment, OK.

**Comment by AlexL777HRO** [comment]
Now, I had uploaded from your source the packet with mecha CMS and under https://mecha-cms.de I see that error:

`Parse error: syntax error, unexpected end of file, expecting function (T_FUNCTION) or const (T_CONST) in /mecha-cms.de/engine/kernel/state.php on line 1`

PHP is 7.4.x and allow url fopen is active.

**Comment by taufik-nurrohman** [comment]
Why it is always line 1?

**Comment by AlexL777HRO** [comment]
So now a new error by the name of:

`Fatal error: Declaration of Page::__toString() must be compatible with File::__toString(): string in /home/mecha-cms.de/lot/x/page/engine/kernel/page.php on line 44`

I have uploaded the mecha again, and I have unchecked the "Minify PHP".

**Comment by AlexL777HRO** [comment]
Ok, now it's all fine, and I have removed this site from Cloudflare, because it's not worked with Cloudflare. I think, that is by the side of Cloudflare, you know, Tovic.

In the next a couple of hours will be the DNS done and after that we will know more about my Mecha CMS website :-).

**Comment by AlexL777HRO** [comment]
The new error is from Firefox browser and I don't know, what is that:

`Content-Encoding-error` in the last version of the browser.

**Comment by taufik-nurrohman** [comment]
It shows content encoding error because there&rsquo;s an error when printing the response. When this happen, usually a log file will be available as `.\engine\log\error`, `.\engine\log\error-x` or `.\engine\log\error-y`.

**Comment by taufik-nurrohman** [comment]
Open `./index.php` then change the `TEST` constant value to `false`:

~~~ php
define('TEST', false);
~~~

This does not happen with PHP 8.x.x though.

**Comment by AlexL777HRO** [comment]
I have uploaded old files of mecha CMS of the version 2.6.3 to mecha-cms.de and no in the backend I have shown that error:

`[30-Oct-2021 07:31:17 Europe/Berlin] PHP Deprecated: stripos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior in ./lot/x/search/engine/r/route.php on line 1`

How I can fix this? PHP is 7.4.x. Many thanks for the help, Tovic!

**Comment by taufik-nurrohman** [comment]
Please download the uncompressed version of [mecha-cms/x.search](https://github.com/mecha-cms/x.search) extension so that I can know exactly where the code line is.

**Comment by AlexL777HRO** [comment]
Good morning, Tovic,

now it's everything fine with my mecha website. I have now uploaded your mecha version 2.6.3 with the default layout, which working fine. And thanks for all the help to me!

**Comment by taufik-nurrohman** [comment]
Yeah, sorry about that. Version 3.0.0 is not yet stable and not ready for production. But once it stable, there will be more extensions and layouts available.

**Comment by taufik-nurrohman** [comment]
@AlexL777HRO could you try again the `start.php` file with the latest update?

**Comment by AlexL777HRO** [comment]
Hi, ok, thank you I will check it today at later evening :-)
